aicsimageio/aics_image.py

```diff
@@ -461,10 +461,16 @@ def get_channel_names(self, scene: int = 0) -> List[str]:
         channels_names: List[str]
             List of strings representing the channel names.
         """
-        try:
-            return self.reader.get_channel_names(scene)
-        except AttributeError:
-            return [str(i) for i in range(self.size_c)]
+        # Get channel names from reader
+        channel_names = self.reader.get_channel_names(scene)
+
+        # Unlike the readers, AICSImage objects always have a channel dimension
+        # In the case the base reader returns None, return a list of "0"
+        if channel_names is None:
+            return ["0"]
```

Comment:
can this be more like `[str(i) for i in range(num_channels)]` ?

Reply:
we should be able to assert that channel_names contains the actual number of channels, in tests

Reply:
Can do. But if this statement was hit we know that the base reader didn't have a channel dimension so was just shorting that loop.